src/Razor/src/Microsoft.VisualStudio.LanguageServerClient.Razor/ProjectRazorJsonPublisher.cs

```diff
@@ -286,15 +286,15 @@ protected virtual void SerializeToFile(ProjectSnapshot projectSnapshot, string p
             {
                 var projectRazorJson = new ProjectRazorJson(publishFilePath, projectSnapshot);
                 _serializer.Serialize(writer, projectRazorJson);
+            }
 
-                var fileInfo = new FileInfo(publishFilePath);
-                if (fileInfo.Exists)
-                {
-                    fileInfo.Delete();
-                }
+            var fileInfo = new FileInfo(publishFilePath);
+            if (fileInfo.Exists)
+            {
+                fileInfo.Delete();
             }
 
-            tempFileInfo.MoveTo(publishFilePath);
+            File.Move(tempFilePath, publishFilePath);
```

Comment:
Note that this matches other code in our repo:
https://github.com/dotnet/razor-tooling/blob/main/src/Razor/src/Microsoft.AspNetCore.Razor.OmniSharpPlugin/DefaultProjectChangePublisher.cs#L104